nautobot/core/celery/task.py

```diff
@@ -4,12 +4,47 @@
 from celery.result import EagerResult
 from celery.utils.functional import maybe_list
 from celery.utils.nodenames import gethostname
+from django.conf import settings
+from django.contrib.auth import get_user_model
 from kombu.utils.uuid import uuid
 
+from nautobot.core.branching import BranchContext
+
 
 class NautobotTask(Task):
     """Nautobot extensions to tasks for integrating with Job machinery."""
 
+    def before_start(self, task_id, args, kwargs):
+        User = get_user_model()
+        user_id = self.request.properties.get("nautobot_job_user_id", None)
+        self._nautobot_branch_context = BranchContext(
+            branch_name=self.request.properties.get("nautobot_job_branch_name", None),
+            user=User.objects.get(id=user_id) if user_id else None,
+        )
+        self._nautobot_branch_context.__enter__()
```

Comment:
Should we call `super().before_start()` here and `super().after_return()` below?

Reply:
IIRC the base celery Task class has no-ops for those, but it wouldn't hurt. Will do.